codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/RuntimeClientPlugin.java

```diff
@@ -41,37 +41,39 @@
 public final class RuntimeClientPlugin implements ToSmithyBuilder<RuntimeClientPlugin> {
 
     private final Symbol resolveFunction;
-    private final ApplyPlugin buildMiddlewareStack;
+    private final ApplyPlugin writeMiddlewareHelpers;
     private final BiPredicate<Model, ServiceShape> servicePredicate;
     private final OperationPredicate operationPredicate;
     private final Set<ConfigField> configFields;
+    private final MiddlewareRegistrar registerMiddleware;
```

Comment:
All that's needed is a symbol pointing to the registration function and a boolean indicating whether it needs to be passed the config. You could default the boolean to true if the plugin has config fields.